### PR TITLE
store: don't discard error body from request device session call

### PR DESCRIPTION
**Pull request by emgee**
I'm sure this could be improved by attempting to parse the error but this is better than discarding it and losing all context.

### CODE REVIEW

**Comment by emgee** [comment]
Spread test failure looks unrelated.

File: store/auth.go [hunk 1]
```diff
@@ -298,7 +299,8 @@ func RequestDeviceSession(serialAssertion, sessionRequest, previousSession strin
 
 	// check return code, error on anything !200
 	if resp.StatusCode != 200 {
-		return "", fmt.Errorf(errorPrefix+"store server returned status %d", resp.StatusCode)
+		body, _ := ioutil.ReadAll(resp.Body) // do our best to read the body
```
Comment (chipaca) [inline store/auth.go]: ummmm... could you instead read up to 4k or 1MB something large-but-capped?

**Comment by chipaca** [comment]
:+1: except for not making it nom all the rams.

**Comment by chipaca** [comment]
A super-:+1: from me

File: store/auth.go (hunk 1 above)
Comment (emgee) [inline store/auth.go]: Fixed using io.LimitedReader and ioutil.ReadAll, as discussed.

**Comment by mvo5** [comment]
Yay, lets merge once tests are green.

**Comment by zyga** [comment]
LGTM

**Pull request merged**